File: 백준/Silver/1059. 좋은 구간/좋은 구간.js
const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")

const l = Number(input[0])
const S = input[1]
  .split(" ")
  .map(Number)
  .sort((a, b) => a - b)
const n = Number(input[2])

let maxIndex = S.findIndex((value) => value > n)
let max = S[maxIndex] // 포함되면 안됨
let min = S[maxIndex - 1] || 0 // 포함되면 안됨

let count = (max-n) * (n-min) -1
console.log(count <=0 ? 0 : count)

File: 백준/Silver/1059. 좋은 구간/좋은 구간.py
import sys
input = sys.stdin.readline

l = int(input())
S = list(map(int, input().split()))
S.sort()
n = int(input())

# n보다 큰 첫 번째 값의 인덱스 찾기
max_index = next((i for i, value in enumerate(S) if value > n), len(S))
max_val = S[max_index] if max_index < len(S) else float('inf')
min_val = S[max_index - 1] if max_index > 0 else 0

count = (max_val - n) * (n - min_val) - 1
print(0 if count <= 0 else count)
